appsec/appsec_test.go

```diff
@@ -0,0 +1,156 @@
+package appsec_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/appsec"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/mocktracer"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+)
+
+func TestTrackUserLoginSuccessEvent(t *testing.T) {
+	t.Run("nominal", func(t *testing.T) {
+		mt := mocktracer.Start()
+		defer mt.Stop()
+
+		span, ctx := tracer.StartSpanFromContext(context.Background(), "example")
+		appsec.TrackUserLoginSuccessEvent(ctx, "user id", map[string]string{"region": "us-east-1"}, tracer.WithUserName("username"))
+		span.Finish()
+
+		// Check the span contains the expected tags.
+		require.Len(t, mt.FinishedSpans(), 1)
+		finished := mt.FinishedSpans()[0]
+		expectedEventPrefix := "appsec.events.users.login.success."
+		require.Equal(t, true, finished.Tag(expectedEventPrefix+"track"))
+		require.Equal(t, "user id", finished.Tag("usr.id"))
+		require.Equal(t, "us-east-1", finished.Tag(expectedEventPrefix+"region"))
+		require.Equal(t, "username", finished.Tag("usr.name"))
+	})
+
+	t.Run("nominal", func(t *testing.T) {
```

Comment:
Maybe differentiate the name from the previous test case, even if it's just nominal-1/nominal-2?

Reply:
> that's what go does automatically: nominal#1 and nominal#2

Oh, okay, good to know :smile: